biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -418,6 +433,237 @@ end subroutine get_harvest_rate_area
+                    available_patch_c =  available_patch_c + harvestable_cohort_c
+                 else
+                    available_patch_c = available_patch_c + harvestable_cohort_c
+                 end if
+              end if
+           end if
+           currentCohort => currentCohort%shorter
+        end do
+
+        ! judge which category the current patch belong to
+        ! since we have not separated forest vs. non-forest
+        ! all carbon belongs to the forest categories
+        do h_index = 1,hlm_num_lu_harvest_cats
+           if (currentPatch%anthro_disturbance_label .eq. primaryforest) then
+              ! Primary
+              if(hlm_harvest_catnames(h_index) .eq. "HARVEST_VH1") then
+                 harvestable_forest_c(h_index) = harvestable_forest_c(h_index) + harvestable_patch_c
+              end if
+           else if (currentPatch%anthro_disturbance_label .eq. secondaryforest .and. &
+                currentPatch%age_since_anthro_disturbance >= secondary_age_threshold) then
+              ! Secondary mature
+              if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH1") then
+                 harvestable_forest_c(h_index) = harvestable_forest_c(h_index) + harvestable_patch_c
+              end if
+           else if (currentPatch%anthro_disturbance_label .eq. secondaryforest .and. &
+                currentPatch%age_since_anthro_disturbance < secondary_age_threshold) then
+              ! Secondary young
+              if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH2") then
+                 harvestable_forest_c(h_index) = harvestable_forest_c(h_index) + harvestable_patch_c
+              end if
+           end if
+        end do
+        currentPatch => currentPatch%younger
+     end do
+
+   end subroutine get_harvestable_carbon
+
+   ! ============================================================================
+
+   subroutine get_harvest_rate_carbon (patch_anthro_disturbance_label, hlm_harvest_catnames, &
+                 hlm_harvest_rates, secondary_age, harvestable_forest_c, &
+                 harvest_rate, harvest_tag, cur_harvest_tag)
+
+     ! -------------------------------------------------------------------------------------------
+     !
+     !  DESCRIPTION:
+     !  get the carbon-based harvest rates based on info passed to FATES from the boundary conditions in.
+     !  assumes logging_time == true
+
+      ! Arguments
+      real(r8), intent(in) :: hlm_harvest_rates(:) ! annual harvest rate per hlm category
+      character(len=64), intent(in) :: hlm_harvest_catnames(:) ! names of hlm harvest categories
+      integer, intent(in) :: patch_anthro_disturbance_label    ! patch level anthro_disturbance_label
+      real(r8), intent(in) :: secondary_age     ! patch level age_since_anthro_disturbance
+      real(r8), intent(in) :: harvestable_forest_c(:)  ! site level forest c matching criteria available for harvest
+      real(r8), intent(out) :: harvest_rate
```

Comment:
what are units for these two variables? one is a carbon stock and one is an areal rate of harvest?

Reply:
Correct. Will add.